assets/src/application/ui/forms/espressoForm/utils.ts

```diff
@@ -79,3 +80,21 @@ export const evalFieldConditions = (conditions: FieldConditions, formData: AnyOb
 	}
 	return conditionsApply;
 };
+
+/**
+ * Converts yup errors object into RFF error shape
+ * @link https://github.com/jquense/yup#validationerrorerrors-string--arraystring-value-any-path-string
+ * @link https://final-form.org/docs/final-form/types/Config#validation-errors
+ * @param validationSchema
+ * @param values
+ */
+export const yupToFinalFormErrors = async <T>(validationSchema: ObjectSchema, values: T) => {
+	try {
+		await validationSchema.validate(values, { abortEarly: false });
+	} catch (error) {
+		return (error as ValidationError).inner.reduce(
+			(formError, innerError) => setIn(formError, innerError.path, innerError.message),
+			{}
+		);
```

Comment:
can this be written a bit more explicit? not clear what is happening at first glance

Reply:
That's the reason I added the links in doc comment. You can see he shape of the objects from those links.

Reply:
the external links are great but what about a really simple basic before and after example as well?